adds AppIdentityCredentials

bshaffer opened the pull request:
fixes #46

bshaffer commented:
/cc @surferjeffatgoogle @tmatsuo

stanley-cheung commented on src/AppIdentityCredentials.php:
```diff
@@ -0,0 +1,106 @@
+   *   string(10) "1444339905"
+   *  }
+   */
+  public function fetchAuthToken(ClientInterface $client = null)
```
We had been naming this kind of unused variable `$unusedClient` elsewhere in the library. See https://github.com/google/google-auth-library-php/blob/master/src/IAMCredentials.php#L75

stanley-cheung commented on src/AppIdentityCredentials.php:
```diff
@@ -0,0 +1,106 @@
+
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Client;
+use google\appengine\api\app_identity\AppIdentityService;
```
Do we need to include this at the top level `composer.json` file?

bshaffer replied:
We actually don't, because if we do then it conflicts with the library on appengine (which gets loaded automatically).

I'm not sure how to best handle that... I suppose a comment on this line would be appropriate. Or adding a mock class somewhere.

stanley-cheung replied:
So you are saying that if this library is used in App Engine, the `google\appengine` library will be loaded automatically? That is interesting.

But what if the library is used outside of App Engine in a different context? Will this line throw an error?

stanley-cheung replied:
Ah I just saw your comment below and looks like we are going to be fine here.

stanley-cheung commented on src/AppIdentityCredentials.php:
```diff
@@ -0,0 +1,106 @@
+  /**
+   * Implements FetchAuthTokenInterface#fetchAuthToken.
+   *
+   * Fetches the auth tokens using the AppIdentityService it is available.
```
Missing an `if` I believe

bshaffer commented:
Thank you for the thorough review, @stanley-cheung! I have addressed your concerns, and added a mock [AppIdentityService](https://github.com/google/google-auth-library-php/pull/75/files#diff-7bfa85c2e884aa91fc9a7d09ded8a7f1R98) class for when developing outside of App Engine. Really, the best way would be for the user to not even be aware the library exists, as these calls made locally will fail anyway.

However, as long as the developers call `$client->useApplicationDefaultCredentials()` and they define the environment variable `GOOGLE_APPLICATION_CREDENTIALS` as a path to their service account credentials in their dev environment, this will function as expected both locally and in production.
